File: Behavioral/Lab21/C#/Strategies/AddRoomStrategy.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Windows.Input;
using C_.Entities;

namespace C_.TemplateMethods
{
    public class AddRoomStrategy : AbstractStrategy
    {
        public AddRoomStrategy(params string[] args) : base(args)
        {
        }
        public override string Execute()
        {
            var agency = Agency.GetInstance();
            if(_params.Length == 0){
                agency.AddEstate(new Room());
                return "Added empty new Room";
            }
            else if(_params.Length == 5){
                agency.AddEstate(new Room(_params[0], _params[1], Convert.ToDouble(_params[2]), Convert.ToDouble(_params[3]), _params[4]));
                return "Added new Room with args: " + string.Join(", ", _params);
            }
            return "Failed to add new Room with args: " + string.Join(", ", _params);
        }
    }
}

File: Behavioral/Lab21/Typescript/src/Strategies/add_room_strategy.ts
import { Agency } from "../Entities/agency";
import { Apartment } from "../Entities/apartment";
import { Room } from "../Entities/room";
import { ICommand } from "../Interfaces/command";
import { AbstractStrategy } from "./abstract_strategy";

export class AddRoomStrategy extends AbstractStrategy {
    constructor(...params: string[]) {
        super(...params);
    }
    public execute(): string {
        const agency = Agency.GetInstance();
        if (this.Params.length === 0) {
            agency.AddEstate(new Apartment());
            return "Added empty new Room";
        } else if (this.Params.length === 5) {
            agency.AddEstate(new Room(
                this.Params[0],
                this.Params[1],
                +this.Params[2],
                +this.Params[3],
                this.Params[4]
            ));
            return "Added new Room with args: " + this.Params.join(", ");
        }
        return "Failed to add new Room with args: " + this.Params.join(", ");
    }
}